src/query/config/src/outer_v0.rs

```diff
@@ -307,6 +344,91 @@ impl TryInto<InnerStorageConfig> for StorageConfig {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CatalogConfig {
+    #[serde(rename = "type")]
+    ty: String,
+    #[serde(flatten)]
+    hive: CatalogHiveConfig,
+}
+
+impl Default for CatalogConfig {
+    fn default() -> Self {
+        Self {
+            ty: "hive".to_string(),
+            hive: CatalogHiveConfig::default(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Args)]
+#[serde(default = "CatalogHiveConfig::empty")]
+pub struct CatalogHiveConfig {
```

Comment:
Maybe we need different config types for compatible and new catalogs?

I prefer to rename `meta_store_address` into `address` and remove clap support for new catalogs.

Reply:
rename `meta_store_address` in outer config will annoy Clap:
```
Argument names must be unique, but 'address' is in use by more than one argument or group'
```
I'll keep the field's name as `meta_store_address`, but make SerDe use `address` by default and `meta_store_address` as an alias.

Reply:
How about providing two different structs for the old hive config and new catalog configs?

We will not support using clap to configure the new catalog configs, so we don't need to bother `clap::Args` here.

Reply:
In the future, we will remove all clap Args support for complex configs like storage/cache/catalog.